biggraphite/drivers/elasticsearch.py

```diff
@@ -647,18 +650,30 @@ def touch_metric(self, metric_name):
         """See the real Accessor for a description."""
         super(_ElasticSearchAccessor, self).touch_metric(metric_name)
         metric_name = ".".join(_components_from_name(metric_name))
-        metric = self.__get_metric(metric_name)
-        self.__touch_metric(metric.meta.index, metric.uuid)
+        document = self.__get_document(metric_name)
+        self.__touch_document(document)
+
+    def __touch_document(self, document):
+        metric = self._document_to_metric(document)
+        new_index = self.get_index(metric)
+        if new_index == document.meta.index:
+            self.__update_existing_document(document)
+        else:
+            self.create_metric(metric)
 
-    def __touch_metric(self, index, document_id):
+    def __update_existing_document(self, document):
         # TODO: state if we should move the document from its index to
         # the current (today) index
+        index = document.meta.index
+        document_id = document.uuid
+        updated_on = datetime.datetime.now()
         data = {
             "doc": {
-                "updated_on": datetime.datetime.now()
+                "updated_on": updated_on
             }
         }
         self.__update_document(data, index, document_id)
+        document.updated_on = updated_on.replace(microsecond=0).isoformat()
```

Comment:
can you add a code comment to explain `replace(microsecond=0).isoformat()` ? I though you could just put a datetime here.

Reply:
I'll extract a function in utils then